docs/tutorials/channels.md

```diff
@@ -720,470 +1049,481 @@ sequenceDiagram
 
-### Command-line interface
-Most applications will expect to receive arguments over the command-line, eg. to point it at a configuration file.
-This is outside the scope of Malachite, but for the purpose of this tutorial we can use Malachite's test CLI instead of creating our own.
+                if reply
+                    .send(ConsensusMsg::StartHeight(
+                        state.current_height,
+                        state.get_validator_set().clone(),
+                    ))
+                    .is_err()
+                {
+                    error!("Failed to send Decided reply");
+                }
+            }
+```
 
-In our `main.rs` file, we can add the following:
+If, on the other hand, we are not lagging behind but are instead asked by one of
+our peer to help them catch up because they are the one lagging behind,
+then the engine might ask the application to provide with the value
+that was decided at some lower height. In that case, we fetch it from our store
+and send it to consensus.
 
 ```rust
-use eyre::Result;
-
-use malachitebft_test_cli::args::Args;
-
-#[tokio::main]
-async fn main() -> Result<()> {
-    // Install a global error handler
-    color_eyre::install()?;
+            AppMsg::GetDecidedValue { height, reply } => {
+                info!(%height, "Received sync request for decided value");
 
-   // Load command-line arguments and possible configuration file.
-   let args = Args::new();
+                let decided_value = state.get_decided_value(height).await;
+                info!(%height, "Found decided value: {decided_value:?}");
 
-   // More to come
-}
+                let raw_decided_value = decided_value.map(|decided_value| RawDecidedValue {
+                    certificate: decided_value.certificate,
+                    value_bytes: ProtobufCodec.encode(&decided_value.value).unwrap(),
+                });
 
+                if reply.send(raw_decided_value).is_err() {
+                    error!("Failed to send GetDecidedValue reply");
+                }
+            }
 ```
 
-The `malachite-test-cli` crate uses the `clap` library to parse the command-line and provide input parameters like
-`--home` as well as commands like `init`, `start` or `testnet`.
+In order to figure out if we can help a peer that is lagging behind,
+the engine may ask us for the height of the earliest available value in our store.
+
+```rust
+            AppMsg::GetHistoryMinHeight { reply } => {
+                let min_height = state.get_earliest_height().await;
 
-These work similarly to CometBFT: `init` creates initial configuration in `$HOME/.malachite`, `testnet` does the same but
-it creates multiple configurations (make sure `--nodes` is set) for running a local testnet, and `start` runs the node.
+                if reply.send(min_height).is_err() {
+                    error!("Failed to send GetHistoryMinHeight reply");
+                }
+            }
+```
 
-### Logging
```

Comment:
Dedicated logging section might still have been useful, no?

Reply:
Right, I will add it to the command-line section, where it is initialized.